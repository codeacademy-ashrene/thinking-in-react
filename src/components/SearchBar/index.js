import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SearchBar extends Component{
  handleFilterTextChange = (e) => {
    this.props.onFilterTextChange(e.target.value);
  }
  handleInStockChange = (e) => {
    this.props.onInStockChange(e.target.checked)
  }
  render(){
    return (
      <form>
        <input type="text" className="search" placeholder="Search..." value={this.props.filterText} onChange={this.handleFilterTextChange}/>
        <input type="checkbox" className="checkBox" checked={this.props.inStockOnly} onChange={this.handleInStockChange}/>
        { ' '}
        Only show products in stock
      </form>
    );
  }
}

SearchBar.propTypes = {
  filterText: PropTypes.string,
  inStockOnly: PropTypes.bool,
};

SearchBar.defaultProps = {
  filterText: '',
  inStockOnly: false,
}
export default SearchBar;