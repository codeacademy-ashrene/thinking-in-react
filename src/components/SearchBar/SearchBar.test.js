import React from 'react';
import SearchBar from '.';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
describe('SearchBar', () => {
  it('should render product table', () => {
    const tree = renderer.create(<SearchBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('checks the value supplied to checkbox', () => {
    const wrapper = shallow(<SearchBar filterText='' inStockOnly/>);
    expect(wrapper.find('.checkBox').props().checked).toEqual(true)
  });
  it('checks the value supplied to searchbox', () => {
    const wrapper = shallow(<SearchBar filterText='Football' inStockOnly={false}/>);
    expect(wrapper.find('.search').props().value).toEqual('Football')
  })
});