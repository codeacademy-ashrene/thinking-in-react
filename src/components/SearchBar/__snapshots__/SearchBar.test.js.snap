// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render product table 1`] = `
<form>
  <input
    className="search"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
  <input
    checked={false}
    className="checkBox"
    onChange={[Function]}
    type="checkbox"
  />
   
  Only show products in stock
</form>
`;
